src/code42cli/cmds/devices.py

```diff
@@ -206,51 +210,78 @@ def _get_device_info(sdk, device_guid):
     is_flag=True,
     help="""Include device settings in output.""",
 )
+@click.option(
+    "--last-connected-before",
+    type=MagicDate(rounding_func=round_datetime_to_day_start),
+    help=f"Include devices only when the 'lastConnected' field is after provided value. {MagicDate.HELP_TEXT}",
+)
+@click.option(
+    "--last-connected-after",
+    type=MagicDate(rounding_func=round_datetime_to_day_end),
+    help=f"Include devices only when 'lastConnected' field is after provided value. {MagicDate.HELP_TEXT}",
+)
+@click.option(
+    "--created-before",
+    type=MagicDate(rounding_func=round_datetime_to_day_start),
+    help=f"Include devices only when 'creationDate' field is less than provided value. {MagicDate.HELP_TEXT}",
+)
+@click.option(
+    "--created-after",
+    type=MagicDate(rounding_func=round_datetime_to_day_end),
+    help=f"Include devices only when 'creationDate' field is greater than provided value. {MagicDate.HELP_TEXT}",
+)
 @format_option
 @sdk_options()
 def list_devices(
     state,
     active,
-    days_since_last_connected,
-    drop_most_recent,
+    inactive,
     org_uid,
     include_backup_usage,
     include_usernames,
     include_settings,
+    last_connected_after,
+    last_connected_before,
+    created_after,
+    created_before,
     format,
 ):
-    """Outputs a list of all devices."""
+    """Get information about many devices."""
+    if inactive:
+        active = False
     columns = [
         "computerId",
         "guid",
         "name",
         "osHostname",
         "status",
         "lastConnected",
+        "creationDate",
         "productVersion",
         "osName",
         "osVersion",
         "userUid",
     ]
-    devices_dataframe = _get_device_dataframe(
```

Comment:
This table is really large.

I notice that `name` and `osHostname` are often the same thing, one having unicode and one not..

Can we consolidate these?

Reply:
Also, do we need `computerId`?

Reply:
`name` and `osHostname` are often but not always the same. `name` is a user-editable field, whereas `osHostname` is taken straight from the os. If the user doesn't change the name, then the two are the same, but I think in practice it is important to have both available.

Reply:
Good to know. I should clarify that I am only referring to the Table output.. For CSV or JSON, all the data should be present.. People should not be using Table for programmatic reasons, so it is good to make it more like a UI summary.

Reply:
To achieve this, all we need is a Header will a predefined list of props to show in the Table to pass to the formatter.  The formatter already knows to include all the data for other formats, iicr

Reply:
Since we're now using the `DataFrameOutputFormatter` to print these results (for Table, JSON, and CSV), the column filtering isn't done based on formatting, but on the `column` list that is passed to `_get_device_dataframe()`. So they're all going to be the same for now.

I think as we refactor other commands to use DataFrames instead of our own formatters, we should come up with an better internal API to tweak which columns are displayed, as well as offering options to the user to pick which ones they want. Hopefully it can be general enough that it can just be added to each command that uses a DataFrame by just wrapping with a decorator.